src/UserRepository/PdoDatabase.php

```diff
@@ -126,7 +126,10 @@ protected function getUserRoles(string $identity) : array
         $stmt->bindParam(':identity', $identity);
 
         if (! $stmt->execute()) {
-            return [];
+            throw new Exception\RuntimeException(sprintf(
+                "Error when running query from config sql_get_roles: "
+                .implode($stmt->errorInfo(), ' ')
+            ));
```

Comment:
There's a number of reasons that `PDOStatement::execute()` could return a boolean `false`, and an error is only one such possibility; in some databases, it may mean no matching records were found.

Additionally, supplying the PDO error information could potentially spill security sensitive information in situations where exceptions are used to display error messages to users.

I'm not entirely sure this is the right approach.

Pinging @Ocramius and @ezimuel to weigh in here.

Reply:
I think we can throw an Exception here (as we did [here](https://github.com/zendframework/zend-expressive-authentication/blob/master/src/UserRepository/PdoDatabase.php#L122))  without using `PDOStatement::errorInfo()` to avoid exposing potential security information. We can think to add a logger into `zend-expressive-authentication` to log this kind of errors and other notices (e.g. missing authentication for security reason).